File: src/main/scala/za/co/absa/statusboard/model/ServiceConfiguration.scala
package za.co.absa.statusboard.model

import io.circe.generic.JsonCodec
import za.co.absa.statusboard.model.ServiceConfiguration.ServiceConfigurationLinks

@JsonCodec
case class ServiceConfiguration (
  name: String, // Part of composite PK
  env: String, // Part of composite PK
  hidden: Boolean,
  snowID: String,
  description: String,
  maintenanceMessage: String,
  links: ServiceConfigurationLinks,
  statusCheckAction: StatusCheckAction,
  statusCheckIntervalSeconds: Int,
  statusCheckNonGreenIntervalSeconds: Int,
  notificationCondition: NotificationCondition,
  notificationAction: Seq[NotificationAction]
)

object ServiceConfiguration {
  @JsonCodec
  case class ServiceConfigurationLinks(
    home: String,
    snow: String,
    support: String,
    documentation: String,
    github: String
  )
}

File: ui/src/app/models/service-configuration.ts
import { NotificationCondition } from './notification-condition';
import { NotificationAction } from './notification-action';
import { StatusCheckAction } from './status-check-action';

export interface ServiceConfigurationLinks {
  home: string;
  snow: string;
  support: string;
  documentation: string;
  github: string;
}

export interface ServiceConfiguration {
  name: string; // Part of composite PK
  env: string; // Part of composite PK
  hidden: boolean;
  snowID: string;
  description: string;
  maintenanceMessage: string;
  links: ServiceConfigurationLinks;
  statusCheckAction: StatusCheckAction;
  statusCheckIntervalSeconds: number;
  statusCheckNonGreenIntervalSeconds: number;
  notificationCondition: NotificationCondition;
  notificationAction: NotificationAction[];
}
